### DISCUSSION TITLE
Cannot set crash site as an entity

### DISCUSSION THREAD

**Discussion by varun-dubagunta-clubs** [SimDynamX/SpaceTeamsPRO#57] [2024 Lunar SAR Competition Q&A] [answered]
I am running into the following error trying to set the crash site as an entity I can set a waypoint to:

 landingSite: st.Entity = st.GetThisSystem().GetParam(st.VarType.entityRef, "CrashedLunarLander")
                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
RuntimeError: Param Exception in (a ParamMap), Key "CrashedLunarLander": Get Nonexistent Param: GetParam with type: EntityRef.

Is this an incorrect way to resolve this object? If so, what part of the API should I be looking at to resolve non-entity objects?

**Comment by connorjak** [comment, marked as answer]
This is invalid for the competition; the rovers are supposed to search for the crash site then the CompetitionBackend script will automatically sense when the rovers "find" the crash site (For task 1, this is when the "LIDAR" scan picks up the crash site (20m radius). The code is supposed to start with no knowledge of the crash site location. All that you need for Task 1 is provided by functions in EntityTelemetry.py and MissionManagerFuncs.py, as well as the example "user made" code in the top level of the STU_Lunar_SAR repo.

For sims besides the competition, the CrashedLunarLander entityref param would need to be added to that System's instance params in the sim config for this code to work.

**Comment by connorjak** [comment]
One of the EntityTelemetry functions gives you the "LIDAR" output, which is a list of relative locations and radii of detected objects. This will be used more widely in Task 2 with the obstacles added in.
